code/modules/jobs/job_types/assistant/gimmick_assistants.dm

```diff
@@ -0,0 +1,183 @@
+// This file contains the code and outfits for the assistant gimmicks station trait! When active, gives assistants random stupid gear
+/datum/job/assistant/get_outfit()
+	if(!HAS_TRAIT(SSstation, STATION_TRAIT_ASSISTANT_GIMMICKS))
+		return ..()
+
+	else
+		var/static/list/gimmicks = list()
+		if(!LAZYLEN(gimmicks))
+			for(var/datum/outfit/job/assistant/gimmick/gimmick_outfit as anything in subtypesof(/datum/outfit/job/assistant/gimmick))
+				gimmicks[gimmick_outfit] = gimmick_outfit::outfit_weight
+
+		return pick_weight(gimmicks)
+
+/// Parent type of gimmick loadouts for assistants for the functional assistant station traits
+/datum/outfit/job/assistant/gimmick
+	name = "Gimmick Assistant"
+	/// The weight of the outfit to be picked
+	var/outfit_weight = 0
+
+/datum/outfit/job/assistant/gimmick/post_equip(mob/living/carbon/human/equipped, visualsOnly)
+	. = ..()
+
+	// Throw up any hoods
+	var/obj/item/clothing/suit/hooded/hood_suit = locate(/obj/item/clothing/suit/hooded) in equipped
+	if(hood_suit)
+		SEND_SIGNAL(hood_suit, COMSIG_ITEM_UI_ACTION_CLICK) //we commit some tomfoolery
```

Comment:
What is going on here?

Reply:
We fake the players UI interaction, a component intercepts it. Other option is using GetComponent which is worse imo

Reply:
No this is far worse than using GetComponent, IMO this is a case which it's fine to use GetComponent

But it's especially not good to send a signal like this, there's a high likelihood you catch something you didn't intend on catching and cause problems

Reply:
> We fake the players UI interaction, a component intercepts it. Other option is using GetComponent which is worse imo

SIGNALS ARE MEANT TO BE EVENTS NOT PROC CALLS ASSHOLE!!!!!!!!!!!!!!!!

Reply:
events in coding arent real